Assets/RedBlueGames/MulliganRenamer/Editor/Locale/LocaleManager.cs

```diff
@@ -0,0 +1,97 @@
+﻿/* MIT License
+
+Copyright (c) 2019 Murillo Pugliesi Lopes, https://github.com/Mukarillo
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+namespace RedBlueGames.MulliganRenamer
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using UnityEditor;
+    using UnityEngine;
+    using UnityEngine.Events;
+
+    public class LocaleManager
+    {
+        public static LocaleManager Instance = new LocaleManager();
+
+        private string LocaleLanguagesPath = Application.dataPath + "/RedBlueGames/MulliganRenamer/Editor/Locale/Content";
```

Comment:
I think this is my only complaint. This hard coded path will prevent users from being able to move the Mulligan folder around in their project, like a plugins or tools folder (Assets/AssetStoreTools/Mulligan)

Instead I would recommend putting them into a Resources folder as a subdirectory in the Mulligan/Editor folder. Maybe Mulligan/Editor/Resources/MulliganLanguages or /MulliganLocale. I'd really like just Resources/Languages but I believe all the directories and assets immediately under Resources folders all get searched, so we should make it somewhat unique.

```
// Loads all text files in a Resources/Languages/ directory,
// which we've placed inside Mulligan/Editor/Resources/MulliganLanguages
var languageJSONs = Resources.LoadAll<TextAsset>("MulliganLanguages/");
foreach (var json in languageJSONs)
{
    var language = JsonUtility.FromJson<LocaleLanguage>(json.text);
    allLanguages.Add(language);
}
```

Reply:
I am not so sure about the Resources folder, all the content inside it will be included in the build. And since this is an Editor extension, I wouldn't like to have this unnecessary files inside the final binary.

I also don't like the fact that is hard coded and spent some time thinking how could I not be, but I couldn't figure it out. Do you have any ideas or suggestions on how to "fix" this?

Reply:
My understanding is that since it would be under an Editor folder it
wouldn't be included in a build. Old documentation suggests that's true
(last sentence under Resources section0:
https://docs.unity3d.com/530/Documentation/Manual/SpecialFolders.html, but
I didn't do a thorough test.

Only other thing I can think is to search for it, or add an option to move
it (this is what DoTween does). But those are both a pain. So my vote is an
Editor/Resources folder.

Reply:
Yeah, you are right, here is what I found in the documentation:

> Note that if the Resources folder is an Editor subfolder, the Assets in it are loadable from Editor scripts but are stripped from builds.

But I will have to change a bit how it works, because we can't get the path of the folder, since can be anywhere in the project and I am getting all the files inside a folder.

Reply:
Ok, that was stupid haha.
I got it working with a Resources folder inside MulliganRenamer/Editor folder.